### PR TITLE
NXP-32714: add signout butn

### PR DESCRIPTION
**Pull request by rakeshkumar1019**
https://github.com/nuxeo/nuxeo-admin-console-ui/assets/108446323/f4dc4a9c-ebaa-411d-a733-96fce98e9276

### CODE REVIEW

File: nuxeo-admin-console-web/angular-app/src/app/auth/services/auth.service.ts [hunk 1]
```diff
@@ -3,28 +3,49 @@ import { HttpClient } from "@angular/common/http";
...
+  constructor(private http: HttpClient,private nuxeoJsClientService: NuxeoJSClientService) {}
 
-  getUser(response: AuthResponseInterface): UserInterface {
-    return response.user;
-  }
 
   getCurrentUser(): Observable<UserInterface> {
-    const url = environment.apiUrl + "/user.json";
-    return this.http.get<AuthResponseInterface>(url).pipe(map(this.getUser));
+    const url = `${this.nuxeoJsClientService.getApiUrl()}` + "me";
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/auth/services/auth.service.ts:18]: included endpoint inside string interpolation

File: nuxeo-admin-console-web/angular-app/src/app/auth/services/auth.service.ts [hunk 2]
```diff
@@ -3,28 +3,49 @@ import { HttpClient } from "@angular/common/http";
...
 
   sso(data: HylandSSORequestInterface): Observable<UserInterface> {
     const url = environment.apiUrl + "/users/sso";
     return this.http
-      .post<AuthResponseInterface>(url, data)
+      .post<AuthUserResponseInterface>(url, data)
       .pipe(map(this.getUser));
   }
+  signOut(): Observable<void> {
+    const url = `${this.nuxeoJsClientService.getBaseUrl()}`+ "logout";
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/auth/services/auth.service.ts:44]: included endpoint inside string interpolation

File: nuxeo-admin-console-web/angular-app/src/app/auth/store/effects.ts [hunk 3]
```diff
@@ -42,12 +37,13 @@ export const ssoEffect = createEffect(
...
     persistenceService = inject(PersistenceService)
   ) => {
+    
     return actions$.pipe(
       ofType(authActions.sso),
       switchMap(({ request }) => {
         return authService.sso(request).pipe(
           map((currentUser: UserInterface) => {
-            persistenceService.set("accessToken", currentUser.token);
+            // persistenceService.set("accessToken", currentUser.token);
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/auth/store/effects.ts:46]: Remove comments

File: nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts [hunk 4]
```diff
@@ -1,8 +1,56 @@
...
+import { authActions } from '../../auth/store/actions';
+import { Observable, Subscription } from 'rxjs';
+import { AuthStateInterface } from '../../auth/types/authState.interface';
+import { UserInterface } from '../../shared/types/user.interface';
+import {NuxeoJSClientService} from "../../shared/services/nuxeo-js-client.service";
 @Component({
-  selector: "header-bar",
-  templateUrl: "./header-bar.component.html",
-  styleUrls: ["./header-bar.component.scss"],
+  selector: 'header-bar',
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts:9]: Double quotes should be used by default

File: nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.html [hunk 5]
```diff
@@ -1,3 +1,4 @@
+<!-- eslint-disable @angular-eslint/template/no-negated-async -->
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.html:1]: Check if this is required

File: nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts [hunk 6]
```diff
@@ -1,8 +1,56 @@
...
+  }
+
+  onSignOut(): void {
+    this.store.dispatch(authActions.signOut());
+    this.redirectToHome();
+  }
+
+  private redirectToHome(): void {
+    const _baseURL = this.nuxeoJsClientService.getBaseUrl();
+    window.location.href = _baseURL + 'login.jsp';
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts:41]: /login.jsp?requestedUrl=nuxeoadmin

File: nuxeo-admin-console-web/angular-app/src/app/shared/types/user.interface.ts [hunk 7]
```diff
@@ -1,9 +1,19 @@
+// export interface UserInterface {
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/shared/types/user.interface.ts:1]: Remove constants

**Review by madhurkulshrestha-hyland — DISMISSED** [review]

**Review by Nishant0928 — APPROVED** [review]

File: nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts [hunk 8]
```diff
@@ -1,8 +1,56 @@
...
+  ngOnDestroy(): void {
+    this.currentUserSubscription.unsubscribe();
+  }
+
+  onSignOut(): void {
+    this.store.dispatch(authActions.signOut());
+    this.redirectToHome();
+  }
+
+  private redirectToHome(): void {
```
Comment (madhurkulshrestha-hyland) [inline nuxeo-admin-console-web/angular-app/src/app/layouts/header-bar/header-bar.component.ts:39]: redirectToLogin

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by swarnadipa-dev — APPROVED** [review]

**Pull request merged**